### PR TITLE
Added new metadata fields

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: publication-doi-commons/src/main/java/no/unit/nva/publication/doi/dto/Publication.java [hunk 1]
```diff
@@ -33,22 +38,26 @@
     public Publication(@JsonProperty("id") URI id,
                        @JsonProperty("institution_owner") URI institutionOwner,
                        @JsonProperty("doi") URI doi,
+                       @JsonProperty("doiRequest") DoiRequest doiRequest,
+                       @JsonProperty("modifiedDate") Instant modifiedDate,
```
Comment (norrs) [inline publication-doi-commons/src/main/java/no/unit/nva/publication/doi/dto/Publication.java:42]: Do we have to worry about how objectMapper serialize these?

File: publication-doi-commons/src/main/java/no/unit/nva/publication/doi/dto/DoiRequest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    public static final class Builder {
+
+        private String status;
+        private Instant modifiedDate;
+
+        public Builder() {
+        }
+
+        public Builder withStatus(String status) {
```
Comment (norrs) [inline publication-doi-commons/src/main/java/no/unit/nva/publication/doi/dto/DoiRequest.java:62]: Turn into enum?

File: publication-doi-commons/src/test/java/no/unit/nva/publication/doi/dto/PublicationTest.java [hunk 3]
```diff
@@ -19,6 +20,7 @@
     private static final String EXAMPLE_CONTRIBUTOR_ID = "https://example.net/contributor/id/4000";
     private static final String EXAMPLE_CONTRIBUTOR_NAME = "Brinx";
     private static final String EXAMPLE_CONTRIBUTOR_ARPID = "989114";
+    public static final String EXAMPLE_DOI_REQUEST_STATUS = "APPROVED";
```
Comment (norrs) [inline publication-doi-commons/src/test/java/no/unit/nva/publication/doi/dto/PublicationTest.java:23]: private

File: publication-doi-commons/src/main/java/no/unit/nva/publication/doi/dto/DoiRequest.java (hunk 2 above)
Comment (janbj) [reply publication-doi-commons/src/main/java/no/unit/nva/publication/doi/dto/DoiRequest.java]: fixed

File: publication-doi-commons/src/test/java/no/unit/nva/publication/doi/dto/PublicationTest.java (hunk 3 above)
Comment (janbj) [reply publication-doi-commons/src/test/java/no/unit/nva/publication/doi/dto/PublicationTest.java]: fixed

File: publication-doi-commons/src/main/java/no/unit/nva/publication/doi/dto/Publication.java (hunk 1 above)
Comment (janbj) [reply publication-doi-commons/src/main/java/no/unit/nva/publication/doi/dto/Publication.java:42]: Don't think so, the class has json properties defined.

**Review by norrs — APPROVED** [review]

**Pull request merged**